packages/compiler/core/checker.ts

```diff
@@ -742,6 +745,79 @@ export function createChecker(program: Program): Checker {
     return getMergedSymbol(sym);
   }
 
+  function resolveCompletions(identifier: IdentifierNode): string[] {
+    let resolveDecorator = false;
+
+    // Walk to first non-MemberExpression parent of identifier. If it's
+    // a TypeReference or DecoratorExpression, then we can complete it.
+    loop: for (let node = identifier.parent; node; node = node!.parent) {
```

Comment:
Feels like we could seperate that loop into a dedicated function `findParentNonMememberExpression(node: Node): {resolveDecorator: boolean, parent: Node}` to be a bit cleaner and remove the need for that label.

Reply:
Fixed.